opencodelists/forms.py

```diff
@@ -89,22 +89,27 @@ def clean_csv_data(self):
 
 
 def validate_csv_data_codes(coding_system, codes):
-    unknown_codes_and_ixs = [
-        (ix, code)
-        for ix, code in enumerate(codes)
-        if code not in set(coding_system.lookup_names(codes))
-    ]
-
-    if unknown_codes_and_ixs:
-        line = unknown_codes_and_ixs[0][0] + 1
-        code = unknown_codes_and_ixs[0][1]
-        if len(unknown_codes_and_ixs) == 1:
-            msg = f"CSV file contains 1 unknown code ({code}) on line {line}"
-        else:
-            num = len(unknown_codes_and_ixs)
-            msg = f"CSV file contains {num} unknown code -- the first ({code}) is on line {line}"
-        msg += f" ({coding_system.short_name} release {coding_system.release_name}, valid from {coding_system.release.valid_from})"
-        raise forms.ValidationError(msg)
+    # Fully implemented codings systems have a `lookup_names` method that is used to
+    # validate the codes in the CSV upload.  However, we also support uploads for some
+    # coding systems that we don't maintain data for (e.g. OPCS4, ReadV2).  Skip code
+    # validation for these systems, and just allow upload of the CSV data as it is.
+    if coding_system.has_database:
```

Comment:
I might be tempted to make this a guard clause and then you don't need to indent everything i.e.
```py
if not coding_system.has_database:
    return
```

But that's ultimately a matter of taste.